telethon/client/messages.py

```diff
@@ -1025,14 +1025,30 @@ def get_key(m):
                 force_document=force_document)
 
         if isinstance(entity, types.InputBotInlineMessageID):
-            return await self(functions.messages.EditInlineBotMessageRequest(
-                id=entity,
-                message=text,
-                no_webpage=not link_preview,
-                entities=msg_entities,
-                media=media,
-                reply_markup=self.build_reply_markup(buttons)
-            ))
+            request = functions.messages.EditInlineBotMessageRequest(
+                    id=entity,
+                    message=text,
+                    no_webpage=not link_preview,
+                    entities=msg_entities,
+                    media=media,
+                    reply_markup=self.build_reply_markup(buttons)
+                )
+            # `EditInlineBotMessageRequest`s have to be sent to a specific datacenter
+            # If the dc_id isn't the same as the currently logged in client, borrow a sender
+            # and send it to the specified datacenter
+            sender = None
+            exported = self.session.dc_id != entity.dc_id
+            if exported:
+                try:
+                    sender = await self._borrow_exported_sender(entity.dc_id)
+                    response = await sender.send(request)
```

Comment:
```suggestion
                    return await sender.send(request)
```

No need for the temporary variable.